pkg/cloudprovider/providers/gce/gce_instancegroup.go

```diff
@@ -130,55 +121,24 @@ func (gce *GCECloud) RemoveInstancesFromInstanceGroup(name string, zone string,
 		&compute.InstanceGroupsRemoveInstancesRequest{
 			Instances: instances,
 		}).Do()
-
 	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
```

Comment:
Consistency with the other functions. Sometimes the controller needs to know if the resource was not found. Checking the error for NotFound should be the responsibility of all consumers. For many calls, consumers were already checking the error for NotFound for a second time.